docs/csi-volume-expansion.md

```diff
@@ -0,0 +1,191 @@
+# Expanding CSI persistent volumes with Kubernetes
+
+## Introduction
+
+The Pure Service Orchestrator Kubernetes CSI driver includes support for CSI volume expansion starting with version 5.2.0.
+Check out our csi driver below version 6.0 [here](https://github.com/purestorage/helm-charts/tree/master/pure-csi).
+
+The feature allows Kubernetes end-users to expand a persistent volume (PV) after creation by resizing a dynamically provisioned persistent volume claim (PVC).
+The end-users are able to edit the `allowVolumeExpansion` boolean flag in Kubernetes `StorageClass` to modify the permission whether a PVC resizing is allowed. 
+
+## Prerequisites
+
+The following prerequisites must be true to ensure the function can work properly:
+
+* Only dynamically provisioned PVCs can be resized.
+* Only allow volume size expansion, shrinking a volume is not allowed.
+* The `StorageClass` that provisions the PVC must support resize. `allowVolumeExpansion` flag is set to true by default in all PSO `StorageClass` since version 5.2.0.
+* The PVC `accessMode` must be `ReadWriteOnce` or `ReadWriteMany`.
+
+### Note
+
+Any PVC creted using a StorageClass where the `parameters.backend=block` will only be resized upon a (re)start of the pod bound to the PVC. 
+
+## Dependencies
+
+* Kubernetes already running, deployed, configured, etc.
+* Kubernetes supports CSI Volume Expansion as beta since 1.16 and GA since Kubernetes 1.19, so ensure you have Kubernetes 1.16+ installed. [More Info](https://kubernetes-csi.github.io/docs/volume-expansion.html)
+* PSO correctly installed and using [Pure CSI Driver v5.2.0](https://github.com/purestorage/helm-charts/releases/tag/5.2.0)+. Our 6.0+ releases are in a different repository [here](https://github.com/purestorage/pso-csi).
+
+
+## Example usages
+
+PSO CSI driver supports `ONLINE` volume expansion capability, i.e. expanding an in-use PersistentVolumeClaim.
+For more details check the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) please. 
+ 
+### FlashArray StorageClass "pure-block" volume expansion
+
+#### 1. Ensure `allowVolumeExpansion` is set to `true` in `pure-block` StorageClass:
```

Comment:
will change. should be true by default.

Reply:
but at least need to mention here, as it's possible this flag is set to false manually.

Reply:
If we're still going to mention it, we should change it to not just say `pure-block`, maybe something like "Ensure `allowVolumeExpansion` is set to `true` in the StorageClass being using:" just to remove any confusion. Probably should do something similar in the header above it ("FlashArray StorageClass "pure-block" volume expansion")

Reply:
> If we're still going to mention it, we should change it to not just say `pure-block`, maybe something like "Ensure `allowVolumeExpansion` is set to `true` in the StorageClass being using:" just to remove any confusion. Probably should do something similar in the header above it ("FlashArray StorageClass "pure-block" volume expansion")

why should we remove the StorageClass name here? are those `pure-block` and `pure-file` name potentially be changed?

Reply:
It's because the customer can also create their own storage classes, so I want to make it obvious that it doesn't just have to be our defaults that support volume expansion. Also, if we do that, we could remove the mostly duplicate `pure-file` section and merge it with this, so we could just say the last step is "if you're using a block backend, you also must restart the pod"

Reply:
oh, just realized this is an example usage section, not liking a tutorial, we are providing 2 specific example use cases here, and the yaml files are provided, so people are able to just copy and run. So I think it's good to keep the storage class name here, but mention if you are using other storage class other than this, please make sure `allowVolumeExpansion` is set to true, how about this?

Reply:
Oh, in that case, yeah that sounds perfect! Thanks for looking into that.